includes/admin/class-wc-rest-woopay-session-controller.php

```diff
@@ -74,12 +79,31 @@ public function check_permission() {
 		return $this->is_request_from_woopay() && $this->has_valid_request_signature();
 	}
 
+	/**
+	 * Validates the cart token value.
+	 *
+	 * @param string|null $cart_token The cart token to validate.
+	 * @return string The validated cart token.
+	 * @throws InvalidArgumentException If the cart token is missing or invalid.
+	 */
+	public function validate_cart_token( $cart_token ): string {
+		if ( ! $cart_token ) {
+			throw new InvalidArgumentException( 'Missing cart token.' );
+		}
+
+		if ( ! JsonWebToken::validate( $cart_token, '@' . wp_salt() ) ) {
```

Comment:
As previously mentioned, we may need to ensure the `JsonWebToken` class exists before we use it.

Alternatively, instead of ensuring the class exists in both places, we could rename this function to `get_validated_cart_token_payload( $cart_token )` and do something like the following:
```php
public function get_session_data( WP_REST_Request $request ): WP_REST_Response {
  $payload = $this->get_validated_cart_token_payload( $request->get_header( 'cart_token' ) );
  $user_id = (int) $payload->user_id ?? null;
  // ...
}
// ...
public function get_validated_cart_token_payload( $cart_token ): string {
  if ( ! $cart_token ) {
    throw new InvalidArgumentException( 'Missing cart token.' );
  }

  if ( ! class_exists( JsonWebToken::class ) ) {
    // TODO: Throw an error.
  }

  if ( ! JsonWebToken::validate( $cart_token, '@' . wp_salt() ) ) {
    throw new InvalidArgumentException( 'Invalid cart token.' );
  }

  return JsonWebToken::get_parts( $cart_token )->payload;
}
//...
```

What are your thoughts on this approach?

Reply:
> To avoid a potentially fatal error

I LOVE this and something I've been wanting to emphasize more often in our PR reviews. I like the abstraction here as well, but with that same thought in mind, noting that we're throwing here I'd lean towards catching on our own throws as well to avoid https://github.com/Automattic/woopay/pull/2655 (particularly with how REST requests can be called).

Reply:
That's a good call.

If `get_validated_cart_token_payload ()` throws, we should catch the throw from `get_session_data()` and return an appropriate HTTP response 👍

Reply:
Good calls! I have refactored this in https://github.com/Automattic/woocommerce-payments/pull/8645/commits/873cc763070436e2005224e6dfae8562bfccaaf6.

Here are the instructions to test the errors don't trigger fatals:

**Pre-requisite**

Comment [this line](https://github.com/Automattic/woocommerce-payments/blob/873cc763070436e2005224e6dfae8562bfccaaf6/includes/admin/class-wc-rest-woopay-session-controller.php#L45). This is just so we can ping the endpoint without any constraints.

**Test: Missing cart token error**

1. Run `curl --location 'http://localhost:8082/wp-json/wc/v3/woopay/session'`
2. It should return `{"code":"rest_invalid_param","message":"Missing cart token.","data":{"status":400}}`.

**Test: Invalid cart token error**

1. Run `curl --location 'http://localhost:8082/wp-json/wc/v3/woopay/session' --header 'Cart-Token: 123'`
2. It should return `{"code":"rest_invalid_param","message":"Invalid cart token.","data":{"status":400}}`.

**Test: `JsonWebToken` class doesn't exist**

1. Comment [the JsonWebToken import line](https://github.com/Automattic/woocommerce-payments/blob/873cc763070436e2005224e6dfae8562bfccaaf6/includes/admin/class-wc-rest-woopay-session-controller.php#L12).
2. Run `curl --location 'http://localhost:8082/wp-json/wc/v3/woopay/session' --header 'Cart-Token: 123'`
3. It should return `{"code":"wcpay_server_error","message":"Cannot validate cart token.","data":{"status":500}}`.

Additionally, it'd be nice if you could run the test `Direct checkout works with third-party cookies disabled` again to make sure it's still working as expected.